File: practice_java_program/notpad/LongestSubarraySumKbySlidingWindow.java
public class LongestSubarraySumKbySlidingWindow{

    public static void main(String[] args) {

        int[] arr = {1,2,3,1,1,1,1};
        int k = 3;

        int left = 0, sum = 0, maxLen = 0;
        int startIndex = 0;   // to store start of max subarray

        for (int right = 0; right < arr.length; right++) {

            sum += arr[right];

            while (sum > k) {
                sum -= arr[left];
                left++;
            }

            if (sum == k) {
                if (right - left + 1 > maxLen) {
                    maxLen = right - left + 1;
                    startIndex = left;   // store start
                }
            }
        }

        System.out.println("Longest Length: " + maxLen);

        if (maxLen > 0) {
            System.out.print("Subarray: ");
            for (int i = startIndex; i < startIndex + maxLen; i++) {
                System.out.print(arr[i] + " ");
            }
        }
    }
}

File: python_oriented_interview/notpad/LongestSubarraySumKbySlidingWindow.py
class LongestSubarraySumKbySlidingWindow:
        arr = [1, 2, 3, 1, 1, 1, 1]
        k = 3
        
        left = 0
        current_sum = 0
        max_len = 0
        start = 0
        end = 0
        
        for right in range(len(arr)):
            current_sum += arr[right]
        
            while current_sum > k:
                current_sum -= arr[left]
                left += 1
        
            if current_sum == k:
                if right - left + 1 > max_len:
                    max_len = right - left + 1
                    start = left
                    end = right
        
        print("Length:", max_len)
        print("Subarray:", arr[start:end+1])